docstore/awsdynamodb/dynamo.go

```diff
@@ -613,25 +615,26 @@ func revisionPrecondition(doc driver.Document, revField string) (*expression.Con
 	return &cb, nil
 }
 
-// TODO(jba): use this if/when we support atomic writes.
-func (c *collection) transactWrite(ctx context.Context, actions []*driver.Action, errs []error, opts *driver.RunActionsOptions, start, end int) {
+func (c *collection) transactWrite(ctx context.Context, actions []*driver.Action, errs []error, opts *driver.RunActionsOptions) {
+	if len(actions) == 0 {
+		return
+	}
 	setErr := func(err error) {
-		for i := start; i <= end; i++ {
-			errs[actions[i].Index] = err
+		for _, a := range actions {
+			errs[a.Index] = err
 		}
 	}
 
+	tws := make([]*dyn.TransactWriteItem, 0, len(actions))
 	var ops []*writeOp
-	tws := make([]*dyn.TransactWriteItem, 0, end-start+1)
-	for i := start; i <= end; i++ {
-		a := actions[i]
-		op, err := c.newWriteOp(a, opts)
+	for _, w := range actions {
+		op, err := c.newWriteOp(w, opts)
 		if err != nil {
-			setErr(err)
-			return
+			errs[w.Index] = err
```

Comment:
You've changed the semantics here: this is no longer atomic, since if one newWriteOp fails the others still happen.

Reply:
Sure, let me revert it, this was to handle the cases when the write operation is non-write kind. But I believe it makes to fail here to be semantically correct.

Reply:
Sure, let me revert it. It was added to handle the case if the action is non-write kind which will be somebug in the code. But I got it that it's making it semantically incorrect.